packages/sourcecred/src/plugins/github/relationalView.js

```diff
@@ -47,7 +47,7 @@ export class RelationalView {
   _mapReferences: Map<N.RawAddress, N.ReferentAddress[]>;
   _mapReferencedBy: Map<N.RawAddress, N.TextContentAddress[]>;
 
-  constructor(): void {
+  constructor(repository?: T.Repository): void {
```

Comment:
Why is this optional? Would it be easy to make required?

Reply:
@blueridger Yeah this is because deserialization mechanism reads `RepoEntry` (internal state) but constructor accepts `T.Repository` (and then converts it to `RepoEntry` internally). I will try to turn it into factory/static method so  construction from  `T.Repository` will look like:
```
var rv = RelationalView.fromRepository(repo);
```
Unless you suggest anything else :man_shrugging: